etc/cmake/options.cmake

```diff
@@ -257,6 +257,13 @@ if(OT_OTNS)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_OTNS_ENABLE=1")
 endif()
 
+option(OT_NCP_SPI "enable NCP SPI support")
+if(OT_NCP_SPI)
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_NCP_SPI_ENABLE=1")
+else()
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_NCP_UART_ENABLE=1")
+endif()
```

Comment:
Move to src/ncp/CMakeLists.txt and refer to:
https://github.com/openthread/openthread/blob/master/src/posix/platform/CMakeLists.txt#L46